src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {

```

Comment:
I know it works that way, but you said 
> ParsingContext is a different thing. It says something about the context we're parsing a statement in. These flags say something about the statement itself.

meaning that StatementFlags are descriptive of a syntactic construct, and (unlike ParsingContext) they are not a normative context marker based on what kinds of nodes you are looking for. Yet, they are passed into parseModuleElementOfKind to filter constructs based on a need. From what you said, it seems like it is more appropriate to pass ParsingContext. The descriptive/normative aspects do not match up.